csl-data.json

```diff
@@ -1,431 +1,452 @@
-                "type": "array",
-                "items": {
-                    "type": "string" 
-                } 
-            },
-            "language": {
-                "type": "string" 
-            },
-            "journalAbbreviation": {
-                "type": "string" 
-            },
-            "shortTitle": {
-                "type": "string" 
-            },
-            "author": {
-                "type": "array",
-                "items": {
-                    "id": "name-variable",
-                    "type": [
+    "definitions": {
+        "dateDefinition": {
+            "type": "object",
+            "allOf": [
+                {
+                    "oneOf": [
                         {
                             "properties": {
-                                "family" : {
-                                    "type": "string" 
-                                },
-                                "given" : {
-                                    "type": "string" 
-                                },
-                                "dropping-particle" : {
-                                    "type": "string" 
-                                },
-                                "non-dropping-particle" : {
-                                    "type": "string" 
-                                },
-                                "suffix" : {
-                                    "type": "string" 
-                                },
-                                "comma-suffix" : {
-                                    "type": [
-                                        "string",
-                                        "number",
-                                        "boolean" 
-                                    ] 
-                                },
-                                "static-ordering" : {
-                                    "type": [
-                                        "string",
-                                        "number",
-                                        "boolean" 
-                                    ] 
-                                },
-                                "literal" : {
-                                    "type": "string" 
-                                },
-                                "parse-names" : {
-                                    "type": [
-                                        "string",
-                                        "number",
-                                        "boolean" 
-                                    ] 
-                                } 
+                                "date-parts": {
+                                    "type": "array",
+                                    "items": {
+                                        "type": "array",
+                                        "items": {
+                                            "type": "number"
+                                        },
```

Comment:
Zotero [currently formats](https://github.com/zotero/zotero/issues/1603) years as strings rather than integers during CSL export.

I think upgrading the schema draft is a great enhancement. However, I think it is crucial that backwards incompatible changes are not made in the initial draft change.

Reply:
There are a few things that got more strict that I didn't flag, like setting required fields for date-parts (either a "date-parts", "literal" or "raw" date), but those seemed pretty basic. Agree we should be careful here.